includes/class-amp-theme-support.php

```diff
@@ -411,23 +411,23 @@ public static function ensure_proper_amp_location() {
 			global $wp;
 			$path_args = [];
 			wp_parse_str( $wp->matched_query, $path_args );
-			if ( isset( $path_args[ amp_get_slug() ] ) && '' !== $path_args[ amp_get_slug() ] ) {
+			if ( isset( $path_args[ amp_get_slug() ] ) && '1' !== $path_args[ amp_get_slug() ] ) {
```

Comment:
I don't think this change is quite right. This code here is asking if the URL looks like this `/foo/amp/bar/`, where there is _something_ after the `/amp/` (like for feeds there is `/feed/rss/` and `/feed/atom/`). The `$path_args` is _not_ the query vars from `$_GET` but rather wat was matched from the rewrite rule. So for a request `/foo/amp/bar/`, here `$path_args[ amp_get_slug() ]` would be set to `bar`. If it is any non-empty string value, then this indicates a redirect is needed to avoid serving an infinite URL space.

Reply:
Yes that makes sense. I've made the changes in https://github.com/ampproject/amp-wp/pull/4312/commits/83ec4a869442e89fda638f7bd408f6db7b3458c2.